compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +289,118 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
+      addError(errorMessage);
+      addError(noteOtherLoc);
+      addError(noteDisambiguate);
+    }
+  }
+
+  std::tuple<AstNode*,std::string> Builder::checkAndUpdateConfig(AstNode* ast, pathVecT& pathVec) {
+    AstNode* ret = nullptr;
+    std::string configUsed;
+    auto configs = parsing::configParams(this->context());
+    if (auto var = ast->toVariable()) {
+      if (var->isConfig()) {
+        // inspect pathVec
+        std::string possibleModule;
+        if (pathVec.size() > 1) {
+          for (size_t i = 1; i < pathVec.size(); i++) {
+            possibleModule += pathVec[i].first.str();
+            possibleModule += ".";
+          }
+        } else if (pathVec.size() == 1) {
+          possibleModule = pathVec[0].first.str();
+          possibleModule += ".";
+        }
+
+        // for config vars, check if they were set from the command line
+        for (auto node : configs) {
+          // TODO: How does this work with module prefixes? What if there is a newSymbolPath value?
+          if ((node.first == var->name().str() && var->visibility() != Decl::PRIVATE)
+              || node.first == possibleModule + var->name().str()) {
+            // found a config that was set via cmd line: update the node
+            // handle deprecations
+            if (auto attribs = var->attributes()) {
+              if (attribs->isDeprecated()) {
+                // TODO: Need proper message handling here
+                std::string msg = "'" + var->name().str() + "' was set via a compiler flag";
+                std::cerr << "warning: " + attribs->deprecationMessage().str() << std::endl;
+                std::cerr << "note: " + msg << std::endl;
+              }
+            }
+            auto parser = parsing::Parser::build(context());
+            parsing::Parser* p = parser.get();
+            std::string inputText;
+            // for types, we can't specify a module name when building the input string
+            // and it's important for the parser to see that it's a type and not just a var assignment
+            if (var->kind() == uast::Variable::TYPE) {
+              inputText = "type " + var->name().str() + "=" + node.second +";";
+            } else {
+              inputText = (!node.second.empty()) ? node.first + "=" + node.second +";" : node.first + "=true;";
+            }
+            // TODO: how to handle nested module configs e.g., -sFoo.Baz.bar=10
+            auto parseResult = p->parseString("CompilationConfigs", inputText.c_str());
+            assert(!parseResult.numErrors());
+            auto mod = parseResult.singleModule();
+            assert(mod);
+            if (mod->stmt(0)->isOpCall()) {
+              assert(mod->stmt(0)->toOpCall()->isBinaryOp());
+              ret = mod->children_[0]->children_[1].release();
+            } else if (mod->stmt(0)->isVariable()) {
+              ret = mod->children_[0]->children_[0].release();
```

Comment:
I wouldn't think you'd want to have both OpCall and Variable cases here. It is happening because for the type variables you parse `type someName = someVal` but for other stuff it is `something = somethingElse`. IMO we could reasonably do `type someName = someVal` and `var someName = someVal`. That would make them both be `Variable`.

I see that you aren't just calling `initExpression` here so that you can call `release`. That would deserve some comment but IMO would be better written with `owned<AstNode>` e.g.:

```
auto mod = parseResult.singleModule();
assert(mod);
owned<AstNode> initNode;
if (auto var = mod->stmt(0)->isVariable()) {
  assert(var->initExpression());
  // steal the init expression and leave the children_ storing nullptr for it
  // note that this mutation is OK because the parsed mod is not stored in Context
  initNode = mod->children_[0]->children_[0].back(); 
} else {
  assert(false && "should only be a variable");
}
```